ddtrace/opentelemetry/span.go

```diff
@@ -171,3 +203,115 @@ func toSpecialAttributes(k string, v attribute.Value) (string, interface{}) {
 		return k, v.AsInterface()
 	}
 }
+
+const (
+	httpServerKey    = "http.server.request"
+	httpClientKey    = "http.client.request"
+	operationNameKey = "operation.name"
+)
+
+func remapOperationName(spanKind oteltrace.SpanKind, attrs map[string]interface{}) string {
+	isClient := spanKind == oteltrace.SpanKindClient
+	isServer := spanKind == oteltrace.SpanKindServer
+
+	// http
+	if _, ok := attrs["http.request.method"]; ok {
+		switch spanKind {
+		case oteltrace.SpanKindServer:
+			return httpServerKey
+		case oteltrace.SpanKindClient:
+			return httpClientKey
+		}
+	}
+
+	// database
+	if v := valueFromAttributes(attrs, "db.system"); v != "" && isClient {
+		return v + ".query"
+	}
+
+	// messaging
+	system := valueFromAttributes(attrs, "messaging.system")
+	op := valueFromAttributes(attrs, "messaging.operation")
+	if system != "" && op != "" {
+		switch spanKind {
+		case oteltrace.SpanKindClient, oteltrace.SpanKindServer,
+			oteltrace.SpanKindConsumer, oteltrace.SpanKindProducer:
+			return system + "." + op
+		}
+	}
+
+	// RPC & AWS
+	rpcValue := valueFromAttributes(attrs, "rpc.system")
+	isRPC := rpcValue != ""
+	isAws := isRPC && (rpcValue == "aws-api")
+	// AWS client
+	if isAws && isClient {
+		if service := valueFromAttributes(attrs, "rpc.service"); service != "" {
+			return "aws." + service + ".request"
+		}
+		return "aws.client.request"
+	}
+	// RPC client
+	if isRPC && isClient {
+		return rpcValue + ".client.request"
+	}
+	// RPC server
+	if isRPC && isServer {
+		return rpcValue + ".server.request"
+	}
+
+	// FAAS client
+	provider := valueFromAttributes(attrs, "faas.invoked_provider")
+	faasName := valueFromAttributes(attrs, "faas.invoked_name")
+	if provider != "" && faasName != "" && isClient {
+		return provider + "." + faasName + ".invoke"
+	}
+
+	//	FAAS server
+	trigger := valueFromAttributes(attrs, "faas.trigger")
+	if trigger != "" && isServer {
+		return trigger + ".invoke"
+	}
+
+	//	Graphql
+	if valueFromAttributes(attrs, "graphql.operation.type") != "" {
+		return "graphql.server.request"
+	}
+
+	// if nothing matches, checking for generic http server/client
+	protocol := valueFromAttributes(attrs, "network.protocol.name")
+	if isServer {
+		if protocol != "" {
+			return protocol + ".server.request"
+		}
+		return "server.request"
+	}
+
+	if isClient {
+		if protocol != "" {
+			return protocol + ".client.request"
+		}
+		return "client.request"
+	}
+
+	if spanKind != oteltrace.SpanKindUnspecified {
+		return spanKind.String()
+	}
+
+	return "otel_unknown"
+}
+
+func valueFromAttributes(attrs map[string]interface{}, key string) string {
```

Comment:
This could also be a method on a `*span` type.

Then you could have something like `s.attribute("faas.trigger")` which would give you the value of the attribute.

It may also be nice for this to return a bool. e.g.

`func (s *span) attribute(key string) (ok bool, val string) {`

This might make lines like 245 a little nicer

Reply:
It will introduce a bunch of `ok` variables, but let's see